### ISSUE TITLE
Incorrect value in request.BodyBytes when body contains non-ASCII values

### ISSUE THREAD

**Issue by mtornwall** [Starcounter/Starcounter#2818]
tl;dr: POST:ing binary data containing values outside the ASCII range to a Starcounter handler produces an incorrect value in `request.BodyBytes`.

Steps to reproduce:

1. Set up a simple post handler like:

        Handle.POST("/post", (Request request) =>
        {
            return 200;
        });

    And set a breakpoint on the return line.

2. POST some binary data to the server using curl or similar:

        $ curl -v -X POST -H "Content-Type: application/octet-stream" http://127.0.0.1:8080/post --data-binary `echo -ne "\xff"`
        * Trying 127.0.0.1...
        * Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
        > POST /post HTTP/1.1
        > User-Agent: curl/7.41.0
        > Host: 127.0.0.1:8080
        > Accept: */*
        > Content-Type: application/octet-stream
        > Content-Length: 1
        >
        * upload completely sent off: 1 out of 1 bytes

As you can see curl reports having sent a single byte. But looking at `request.BodyBytes` in the debugger it contains `{0xef, 0xbf, 0xbd}`, even though `ContentLength` is 1 as expected and `request.Body` looks reasonable. This behavior doesn't manifest itself when sending characters in the ASCII range (e.g., 0x7f).

Affects Starcounter 2.0.3263.3.

**Comment by alemoi** [comment]
Thanks for thorough description! I will take a look at it!

**Comment by alemoi** [comment]
Fixed via 9945436d21dd628fea8f688301f315a67e58d7d7
It was the order how body bytes were created (they were created from the body string if it existed, so I changed that).
However we don't check for correct UTF8 sequences.
Here is an example demonstrating the problem:
```c#
                unsafe {
                    SByte[] oneByte = new SByte[1] { -1 }; // Not a correct UTF8 sequence.
                    fixed (SByte* ptr = oneByte) {

                        // The following string will contain just one UTF8 character.
                        String gg = new String(ptr, 0, 1, Encoding.UTF8);

                        // The converted bytes will be represented by 3 bytes.
                        Byte[] bytes = Encoding.UTF8.GetBytes(gg);
                    }
                }
```
And one more remark, in Windows for Curl to send the binary character the only way I found is through the file:
`curl -v -X POST -H "Content-Type: application/octet-stream" http://127.0.0.1:8080/post --data-binary @binaryfile`
